### PR TITLE
DEVEXP-556: Rebasing k8s.io

### PR DESCRIPTION
**Pull request by ricardomaraschini**
Updates and migrates codebase to use k8s.io 1.18 packages. Most of the changes here are  straightforward (add a `context` and `*Options` to function calls), one has happened due to build constraints (one of our dependencies started to use `errors` package so a bump on go version was needed) and one for convenience (removed a pass by reference on `Delete` functions).

### CODE REVIEW

**Comment by ricardomaraschini** [comment]
/hold

we gonna need to update build to go version 1.13 at least.

**Comment by ricardomaraschini** [comment]
Awaiting for https://github.com/openshift/release/pull/8027 to be merged so tests succeed.

File: go.mod [hunk 1]
```diff
@@ -51,23 +50,19 @@ require (
...
-	google.golang.org/grpc v1.25.1 // indirect
 	gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15 // indirect
-	gopkg.in/yaml.v2 v2.2.7
-	k8s.io/api v0.17.1
-	k8s.io/apimachinery v0.17.1
-	k8s.io/client-go v0.17.1
-	k8s.io/code-generator v0.17.1
-	k8s.io/gengo v0.0.0-20191010091904-7fa3014cb28f // indirect
+	gopkg.in/yaml.v2 v2.2.8
+	k8s.io/api v0.18.0
```
Comment (gabemontero) [inline go.mod:62]: Hey @ricardomaraschini - in the openshift/origin PR they are referencing 1.18.0-rc.1

Were you aware?  Or was there a reason you did not go with that ?

Comment (ricardomaraschini) [reply go.mod:62]: No, I was no aware of it. Is this `rc` a release candidate to 1.18(0.18.0)? Could you point me to the PR you are referring to?

Comment (gabemontero) [reply go.mod:62]: Well, I see when I tried rc.1, subsequent updates for openshift/api,client-go,library-go bumped it to v0.18.0 anyway ;-) 

So *nevermind* :-)

**Pull request merged**